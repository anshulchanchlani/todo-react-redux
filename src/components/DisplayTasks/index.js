import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { toggleTask, deleteTask } from '../../actions'
import Task from '../Task'
import './index.scss'
import { Container, Row, Col } from 'react-grid-system';
class DisplayTasks extends Component {

    render() {
        let doneItems = [], undoneItems = [];
        let self = this;

        this.props.tasks && this.props.tasks.length > 0 && this.props.tasks.forEach(function (item) {

            if (item.completed) {
                doneItems.push(
                    <Task statusOfTask="doneItems" deleteTask={self.props.deleteTask} key={item.id} item={item} handleOnCheck={self.props.toggleTask} />)
            } else {
                undoneItems.push(<Task statusOfTask="undoneItems" deleteTask={self.props.deleteTask} key={item.id} item={item} handleOnCheck={self.props.toggleTask} />)
            }
        })

        return (
            (undoneItems.length > 0 || doneItems.length > 0) && 
            <Container className="displayTaskSection">
                <Row>
                    <Col xs={12} lg={5} md={5} sm={12}>
                        <ul aria-label="Undone Items" className="displayTasksList undoneItemsUl">

                            {undoneItems}
                        </ul>
                    </Col>
                 
                    <Col xs={12} lg={5} md={5} sm={12}>
                        <ul aria-label="Done Items" className="displayTasksList doneItemsUl">

                            {doneItems}
                        </ul>
                    </Col>
                </Row>
            </Container>
        );

}}
function mapDispatchToProps(dispatch){
    return bindActionCreators({toggleTask,deleteTask},dispatch)
}

export default connect(null,mapDispatchToProps)(DisplayTasks)
