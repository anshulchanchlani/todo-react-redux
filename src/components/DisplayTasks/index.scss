.displayTaskSection{
  

   margin:auto;
}
.displayTasksList{  
    list-style-type: none;
    
   
   
    li{
        width:auto;
        height:auto;
        margin-bottom:10px;
        padding:25px;
        border-radius: 10px;
    }
}

.doneItems{
        
        background-color:rgb(156, 224, 156);
   
    }
.undoneItems{
       
        background-color:skyblue;
        
    }
