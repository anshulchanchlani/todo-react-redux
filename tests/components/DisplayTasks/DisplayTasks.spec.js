import React from 'react';
import DisplayTasks from '../../../src/components/DisplayTasks';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import reduxPromise from 'redux-promise'
import {shallow} from 'enzyme';




describe('Display Tasks tests',()=>{
    const wrapper = shallow(<DisplayTasks/>)
    
    test('Display Tasks renders',()=>{
      expect(wrapper).toMatchSnapshot();
    })

    
    
})
