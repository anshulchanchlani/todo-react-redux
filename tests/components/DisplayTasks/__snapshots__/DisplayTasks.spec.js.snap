// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Tasks tests Display Tasks renders 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
